kobo/xmlrpc.py

```diff
@@ -489,16 +486,80 @@ def _single_request(self, host, handler, request_body, verbose=0):
             response.read()
         raise xmlrpclib.ProtocolError(host + handler, response.status, response.reason, response.msg)
 
+    def _single_request3(self, host, handler, request_body, verbose=False):
+        # issue XML-RPC request
```

Comment:
Could you explain a bit, where does this code come from - is it your own invention or was it copy-pasted from somewhere?

Reply:
It's a copy of `_single_request()` modified to match Py3 `xmlrpc.client.Transport` implementation.

Reply:
Thanks for explaining.